src/Behat/Behat/Definition/Pattern/PatternTransformer.php

```diff
@@ -79,6 +79,24 @@ public function transformPatternToRegex($pattern)
         return $this->patternToRegexpCache[$pattern];
     }
 
+    /**
+     * Tests if a pattern matches a given step text.
+     *
+     * If the pattern does not match, returns false. Otherwise, returns an array of the portions of the
+     * step text that match any placeholders within the pattern.
+     *
+     * @return false|array<int|string,string>
+     */
+    public function matchPattern(string $pattern, string $stepText): false|array
```

Comment:
It doesn't 100% feel like this method belongs in a class called `PatternTransformer` but I felt it was an OK trade-off for being able to be certain that the ContextSnippetGenerator is using the exact same logic as the RepositorySearchEngine to check if the pattern matches the stepText.